src/observer.tsx

```diff
@@ -2,9 +2,127 @@ import { FunctionalComponent, defineComponent, SetupContext } from 'vue';
 import { Vue } from 'vue-facing-decorator';
 import { Observer } from 'mobx-vue-lite';
 import { Constructor } from 'web-utility';
+import {
+  IReactionDisposer,
+  IReactionPublic,
+  reaction as mobxReaction
+} from 'mobx';
 
 export type VueInstance = InstanceType<typeof Vue>;
 
+export type ReactionExpression<I = any, O = any> = (
+  data: I,
+  reaction: IReactionPublic
+) => O;
+
+export type ReactionEffect<V> = (
+  newValue: V,
+  oldValue: V,
+  reaction: IReactionPublic
+) => any;
+
+interface ReactionItem {
+  expression: ReactionExpression;
+  effect: (...data: any[]) => any;
+}
+
+const reactionMap = new WeakMap<object, ReactionItem[]>();
+
+/**
+ * Method decorator of [MobX `reaction()`](https://mobx.js.org/reactions.html#reaction)
+ *
+ * @example
+ * ```tsx
+ * import { observable } from 'mobx';
+ * import { Vue, Component, toNative } from 'vue-facing-decorator';
+ * import { observer, reaction } from 'mobx-vue-helper';
+ *
+ * @Component
+ * @observer
+ * class MyTag extends Vue {
+ *     @observable
+ *     accessor count = 0;
+ *
+ *     @reaction(({ count }) => count)
+ *     handleCountChange(newValue: number, oldValue: number) {
+ *         console.log(`Count changed from ${oldValue} to ${newValue}`);
+ *     }
+ *
+ *     render() {
+ *        return (
+ *            <button onClick={() => this.count++}>
+ *                Up count {this.count}
+ *            </button>
+ *        );
+ *    }
+ * }
+ * export default toNative(MyTag);
+ * ```
+ */
+export const reaction =
+  <C extends VueInstance, V>(expression: ReactionExpression<C, V>) =>
+  (
+    effect: ReactionEffect<V>,
+    { addInitializer }: ClassMethodDecoratorContext<C>
+  ) =>
+    addInitializer(function () {
+      const reactions = reactionMap.get(this) || [];
+
+      reactions.push({ expression, effect });
+
+      reactionMap.set(this, reactions);
+    });
+
+function wrapClass<T extends Constructor<VueInstance>>(Component: T): T {
+  class ObserverComponent extends (Component as Constructor<VueInstance>) {
+    protected disposers: IReactionDisposer[] = [];
+
+    constructor(...args: any[]) {
+      super(...args);
+
+      // Wait for next tick to ensure component is fully initialized
+      Promise.resolve().then(() => this.bootReactions());
+    }
+
+    protected bootReactions() {
+      const reactions = reactionMap.get(this) || [];
+
+      this.disposers.push(
+        ...reactions.map(({ expression, effect }) =>
+          mobxReaction(
+            reaction => expression(this, reaction),
+            effect.bind(this)
+          )
+        )
+      );
+    }
+
+    render() {
+      const parentRender = (Component.prototype as any).render as
+        | (() => unknown)
+        | undefined;
+      if (typeof parentRender === 'function') {
+        return <Observer>{() => parentRender.call(this)}</Observer>;
+      }
+      return null;
+    }
```

Comment:
```suggestion
    render() {
      return <Observer>{() => super.render?.()}</Observer>;
    }
```
全文直接用 `super` 啊，不然为什么要用类？